src/main/java/uk/gov/hmcts/darts/event/service/impl/EventMappingServiceImpl.java

```diff
@@ -51,12 +60,28 @@ public EventMapping postEventMapping(EventMapping eventMapping, Boolean isRevisi
             }
             var createdEventHandler = eventHandlerRepository.saveAndFlush(eventHandlerEntity);
 
+            if (isUpdateToExistingMappingRequest(isRevision)) {
+                updatePreviousVersionsToInactive(activeMappings);
+            }
```

Comment:
When trying to revise an existing type/subtype (with the application standing up locally), I get:

```json
{
	"title": "Internal Server Error",
	"status": 500,
	"detail": "could not execute statement [ERROR: duplicate key value violates unique constraint \"event_handler_event_type_event_event_sub_type_unq\"\n  Detail: Key (event_type, event_sub_type)=(101010101010, 10) already exists.] [insert into darts.event_handler (active,created_by,created_ts,event_name,handler,is_reporting_restriction,event_sub_type,event_type,evh_id) values (?,?,?,?,?,?,?,?,?)]; SQL [insert into darts.event_handler (active,created_by,created_ts,event_name,handler,is_reporting_restriction,event_sub_type,event_type,evh_id) values (?,?,?,?,?,?,?,?,?)]; constraint [event_handler_event_type_event_event_sub_type_unq]"
}
```

Can you reproduce that?

The following index constraint exists on the `event_handler` table:

```sql
    create unique index event_handler_event_type_event_event_sub_type_unq
    on event_handler (event_type, event_sub_type)
    where active;
```
So I wonder if we first need to set the prior revisions to inactive before saving the new entity:

```suggestion
            if (isUpdateToExistingMappingRequest(isRevision)) {
                updatePreviousVersionsToInactive(activeMappings);
            }
            
            var createdEventHandler = eventHandlerRepository.saveAndFlush(eventHandlerEntity);
```

 It's interesting this isn't showing up as an issue in the integration test.